v-next/hardhat/src/internal/cli/telemetry/sentry/transport.ts

```diff
@@ -0,0 +1,204 @@
+import type {
+  Envelope,
+  Transport,
+  TransportMakeRequestResponse,
+} from "@sentry/core";
+
+import { spawn } from "node:child_process";
+import { fileURLToPath } from "node:url";
+
+import { createTransport, serializeEnvelope } from "@sentry/core";
+import debug from "debug";
+
+const log = debug("hardhat:core:sentry:transport");
+
+/**
+ * Creates a detached process transport.
+ *
+ * This transport spawns a detached process synchronously and sends the envelope
+ * from there.
+ *
+ * This means that the Hardhat process doesn't have to wait for the request
+ * to finish before exiting, flushing the transport, not closing the client.
+ *
+ * This is meant to be use as THE main transport in Hardhat.
+ *
+ * @param dsn The DSN to use to send the envelope to Sentry.
+ * @param release The release/version of Hardhat.
+ * @param environment The environment of Hardhat.
+ * @param configPath The path to the config file.
+ */
+export function createDetachedProcessTransport(
+  dsn: string,
+  release: string,
+  environment: string,
+  configPath?: string,
+): Transport {
+  return {
+    send: (envelope) => {
+      const verbose = log.enabled;
+
+      // **Synchronously** spawn a detached subprocess here
+
+      const out = verbose ? process.stdout : "ignore";
```

Comment:
I think this is a potential epipe if `DEBUG=*` is on.

I suspect the way of dealing with this is within the subprocess, so a function that only logs if stdout is open, or does `debug` take care of that?

Reply:
To be clear, we can do this later as a follow up PR.

Reply:
It may happen in an extreme case. But given that you need to be running in debug mode, i wouldn't care much about it.